pkg/sidecar/server.go

```diff
@@ -151,13 +153,33 @@ func maxClients(h http.Handler, n int) http.Handler {
 	})
 }
 
+func prepareUrl(svc string, endpoint string) string {
+	if !strings.Contains(svc, ":") {
+		svc = fmt.Sprintf("%s:%d", svc, serverPort)
+	}
+	return fmt.Sprintf("http://%s%s", svc, endpoint)
+}
+
+func fastTimeoutTransport(dialTimeout int) http.RoundTripper {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   time.Duration(dialTimeout) * time.Second,
```

Comment:
I could not find a better way to adjust the timeout for the initial `connect` to an empty service. The problem we are trying to solve: when a k8s service is empty, the connect operation will hang until the timeout is reached.The default being 30s meant that we had to wait 30s before trying the next service.